src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Main program of Duke program.
+ *
+ * @author Yu Heng
+ */
+public class Duke {
+    private final String LINE_BREAK = "\n\t ^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^-^\n";
+    private TaskList taskList;
+    private Storage storage;
+    private Ui ui;
+    private boolean hasGui;
```

Comment:
This is cool, users have the option to choose to use either the CLI or GUI?

Reply:
This is a future feature mostly aimed for developer users/myself to toggle between CLI and GUI. I'm not entirely sure if it's feasible to let users have the option to switch from CLI to GUI and vice versa on the fly.